subsys/bluetooth/mesh/light_ctrl_srv.c

```diff
@@ -173,22 +179,22 @@ static void light_set(struct bt_mesh_light_ctrl_srv *srv, uint16_t lvl,
 	lightness_srv_change_lvl(srv->lightness, NULL, &set, &dummy);
 }
 
-static uint32_t curr_fade_time(struct bt_mesh_light_ctrl_srv *srv)
+static uint32_t remaining_fade_time(struct bt_mesh_light_ctrl_srv *srv)
 {
 	if (!atomic_test_bit(&srv->flags, FLAG_TRANSITION)) {
 		return 0;
 	}
 
-	return srv->fade.duration - k_delayed_work_remaining_get(&srv->timer);
+	return k_ticks_to_ms_ceil32(k_work_delayable_remaining_get(&srv->timer));
 }
 
-static uint32_t remaining_fade_time(struct bt_mesh_light_ctrl_srv *srv)
+static uint32_t curr_fade_time(struct bt_mesh_light_ctrl_srv *srv)
 {
 	if (!atomic_test_bit(&srv->flags, FLAG_TRANSITION)) {
 		return 0;
 	}
 
-	return k_delayed_work_remaining_get(&srv->timer);
+	return srv->fade.duration - remaining_fade_time(srv);
```

Comment:
Might it cause a negative value (as a result to the wrong unsigned value) due to the ceiling within `remaining_fade_time`?